mlflow/pipelines/utils/step.py

```diff
@@ -76,9 +76,21 @@ def display_html(html_data: str = None, html_file_path: str = None) -> None:
         import shutil
         import subprocess
 
-        if os.path.exists(html_file_path) and shutil.which("open") is not None:
+        # Use xdg-open in Linux environment
+        if shutil.which("xdg-open") is not None:
+            open_tool = shutil.which("xdg-open")
+        elif shutil.which("open") is not None:
+            open_tool = shutil.which("open")
+
+        if os.path.exists(html_file_path) and open_tool is not None:
```

Comment:
This code fails when both `open` and `xdg-open` are unavailable.

```suggestion
        if shutil.which("xdg-open") is not None:
            open_tool = shutil.which("xdg-open")
        elif shutil.which("open") is not None:
            open_tool = shutil.which("open")
        else:
            open_tool = None
        # or
        # open_tool = shutil.which("open") or shutil.which("xdg-open")

        if os.path.exists(html_file_path) and open_tool is not None:
```